homeassistant/components/renault/sensor.py

```diff
@@ -0,0 +1,349 @@
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_TEMPERATURE,
+    LENGTH_KILOMETERS,
+    PERCENTAGE,
+    POWER_KILO_WATT,
+    TEMP_CELSIUS,
+    TIME_MINUTES,
+    VOLUME_LITERS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.icon import icon_for_battery_level
+from homeassistant.util import slugify
+
+from .const import (
+    DEVICE_CLASS_CHARGE_MODE,
+    DEVICE_CLASS_CHARGE_STATE,
+    DEVICE_CLASS_PLUG_STATE,
+    DOMAIN,
+)
+from .renault_entities import (
+    RenaultBatteryDataEntity,
+    RenaultChargeModeDataEntity,
+    RenaultCockpitDataEntity,
+    RenaultDataEntity,
+    RenaultHVACDataEntity,
+)
+from .renault_hub import RenaultHub
+from .renault_vehicle import RenaultVehicleProxy
+
+ATTR_BATTERY_AVAILABLE_ENERGY = "battery_available_energy"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Renault entities from config entry."""
+    proxy: RenaultHub = hass.data[DOMAIN][config_entry.unique_id]
+    entities = await get_entities(proxy)
+    async_add_entities(entities)
+
+
+async def get_entities(proxy: RenaultHub) -> list[RenaultDataEntity]:
+    """Create Renault entities for all vehicles."""
+    entities = []
+    for vehicle in proxy.vehicles.values():
+        entities.extend(await get_vehicle_entities(vehicle))
+    return entities
+
+
+async def get_vehicle_entities(vehicle: RenaultVehicleProxy) -> list[RenaultDataEntity]:
+    """Create Renault entities for single vehicle."""
+    entities: list[RenaultDataEntity] = []
+    if "cockpit" in vehicle.coordinators:
+        entities.append(RenaultMileageSensor(vehicle, "Mileage"))
+        if vehicle.details.uses_fuel():
+            entities.append(RenaultFuelAutonomySensor(vehicle, "Fuel Autonomy"))
+            entities.append(RenaultFuelQuantitySensor(vehicle, "Fuel Quantity"))
+    if "hvac_status" in vehicle.coordinators:
+        entities.append(RenaultOutsideTemperatureSensor(vehicle, "Outside Temperature"))
+    if "battery" in vehicle.coordinators:
+        entities.append(RenaultBatteryLevelSensor(vehicle, "Battery Level"))
+        entities.append(RenaultChargeStateSensor(vehicle, "Charge State"))
+        entities.append(
+            RenaultChargingRemainingTimeSensor(vehicle, "Charging Remaining Time")
+        )
+        entities.append(RenaultChargingPowerSensor(vehicle, "Charging Power"))
+        entities.append(RenaultPlugStateSensor(vehicle, "Plug State"))
+        entities.append(RenaultBatteryAutonomySensor(vehicle, "Battery Autonomy"))
+        entities.append(RenaultBatteryTemperatureSensor(vehicle, "Battery Temperature"))
+    if "charge_mode" in vehicle.coordinators:
+        entities.append(RenaultChargeModeSensor(vehicle, "Charge Mode"))
+    return entities
+
+
+class RenaultBatteryAutonomySensor(RenaultBatteryDataEntity):
```

Comment:
Sensors need to extend from SensorEntity, seems that that is not the case in this integration.

Reply:
Implemented